sycl/test/esimd/memory_properties.cpp

```diff
@@ -144,3 +151,60 @@ foo(AccType &acc, float *ptrf, int byte_offset32, size_t byte_offset64) {
   // CHECK: call <16 x float> @llvm.genx.oword.ld.v16f32(i32 0, i32 {{[^)]+}}, i32 {{[^)]+}})
   auto z3 = block_load<float, 16>(acc, byte_offset64, props_c16);
 }
+
+// CHECK-LABEL: define {{.*}} @_Z18test_atomic_update{{.*}}
+SYCL_ESIMD_FUNCTION SYCL_EXTERNAL void
+test_atomic_update(AccType &acc, float *ptrf, int byte_offset32,
+                   size_t byte_offset64) {
+  constexpr int VL = 4;
+  int *ptr = 0;
+  uintptr_t addr = reinterpret_cast<uintptr_t>(ptr);
+
+  simd<uint32_t, VL> offsets = simd<uint32_t, VL>(1) * sizeof(int);
+  auto offsets_view = offsets.select<VL, 1>();
+
+  auto add = simd<int, VL>(5);
+  auto compare = simd<int, VL>(VL, 1);
+  auto swap = compare * 2;
+  auto pred = simd_mask<VL>(1);
+
+  properties props_a{cache_hint_L1<cache_hint::uncached>,
+                     cache_hint_L2<cache_hint::write_back>};
+
+  properties props_b{cache_hint_L1<cache_hint::uncached>,
+                     cache_hint_L2<cache_hint::uncached>};
+
+  // CHECK: call <4 x i32> @llvm.genx.lsc.xatomic.stateless.v4i32.v4i1.v4i64(<4 x i1> {{[^)]+}} i8 8, i8 1, i8 3, i16 1, i32 0, i8 3, i8 1, i8 1, i8 0, <4 x i64> {{[^)]+}}, <4 x i32> undef, <4 x i32> undef, i32 0, <4 x i32> undef)
+  auto res_atomic_0 =
+      atomic_update<atomic_op::inc, int>(ptr, offsets, pred, props_a);
+
+  // CHECK: call <4 x i32> @llvm.genx.lsc.xatomic.stateless.v4i32.v4i1.v4i64(<4 x i1> {{[^)]+}} i8 8, i8 1, i8 1, i16 1, i32 0, i8 3, i8 1, i8 1, i8 0, <4 x i64> {{[^)]+}}, <4 x i32> undef, <4 x i32> undef, i32 0, <4 x i32> undef)
+  auto res_atomic_1 =
+      atomic_update<atomic_op::inc, int>(ptr, offsets, pred, props_b);
+
+  // CHECK: call <4 x i32> @llvm.genx.lsc.xatomic.stateless.v4i32.v4i1.v4i64(<4 x i1> {{[^)]+}} i8 8, i8 1, i8 3, i16 1, i32 0, i8 3, i8 1, i8 1, i8 0, <4 x i64> {{[^)]+}}, <4 x i32> undef, <4 x i32> undef, i32 0, <4 x i32> undef)
+  auto res_atomic_2 = atomic_update<atomic_op::inc, int>(ptr, offsets, props_a);
+
+  // CHECK: call <4 x i32> @llvm.genx.lsc.xatomic.stateless.v4i32.v4i1.v4i64(<4 x i1> {{[^)]+}} i8 8, i8 1, i8 3, i16 1, i32 0, i8 3, i8 1, i8 1, i8 0, <4 x i64> {{[^)]+}}, <4 x i32> undef, <4 x i32> undef, i32 0, <4 x i32> undef)
+  auto res_atomic_3 =
+      atomic_update<atomic_op::inc, int>(ptr, offsets_view, pred, props_a);
+
+  auto res_atomic_4 =
+      atomic_update<atomic_op::inc, int, VL>(ptr, offsets_view, props_a);
+
+  // atomic_upate without cache hints:
+  // CHECK: call <4 x i32> @llvm.genx.svm.atomic.inc.v4i32.v4i1.v4i64(<4 x i1> {{[^)]+}}, <4 x i64> {{[^)]+}}, <4 x i32> undef)
+  auto res_atomic_5 =
+      atomic_update<atomic_op::inc, int, VL>(ptr, offsets, pred);
```

Comment:
Please add one more variant to this test: (ptr, offsets). I.e. without 'pred' and without 'props'.

Reply:
Done in https://github.com/intel/llvm/pull/11833/commits/ffe19702c45355afd8b7ab17d2170cd783e9f670.